src/sentry/static/sentry/app/views/eventsV2/eventDetails/content.tsx

```diff
@@ -0,0 +1,339 @@
+import {getMessage, getTitle} from 'app/utils/events';
+import {Organization, Event} from 'app/types';
+import SentryTypes from 'app/sentryTypes';
+import getDynamicText from 'app/utils/getDynamicText';
+import overflowEllipsis from 'app/styles/overflowEllipsis';
+import DateTime from 'app/components/dateTime';
+import ExternalLink from 'app/components/links/externalLink';
+import FileSize from 'app/components/fileSize';
+import {PageHeader} from 'app/styles/organization';
+import NotFound from 'app/components/errors/notFound';
+import AsyncComponent from 'app/components/asyncComponent';
+
+import EventView from '../eventView';
+import {hasAggregateField, EventQuery} from '../utils';
+import Pagination from './pagination';
+import LineGraph from './lineGraph';
+import RelatedEvents from '../relatedEvents';
+import TagsTable from '../tagsTable';
+import EventInterfaces from '../eventInterfaces';
+import LinkedIssuePreview from '../linkedIssuePreview';
+import DiscoverBreadcrumb from '../breadcrumb';
+
+const slugValidator = function(
+  props: {[key: string]: any},
+  propName: string,
+  componentName: string
+) {
+  const value = props[propName];
+  // Accept slugs that look like:
+  // * project-slug:deadbeef
+  if (value && typeof value === 'string' && !/^(?:[^:]+):(?:[a-f0-9-]+)$/.test(value)) {
+    return new Error(`Invalid value for ${propName} provided to ${componentName}.`);
+  }
+  return null;
+};
+
+type Props = {
+  organization: Organization;
+  location: Location;
+  params: Params;
+  api: Client;
+  eventSlug: string;
+};
+
+type State = {
+  event: Event | undefined;
+};
+
+class EventDetailsContent extends AsyncComponent<Props, State & AsyncComponent['state']> {
+  static propTypes: any = {
+    organization: SentryTypes.Organization.isRequired,
+    eventSlug: slugValidator,
+    location: PropTypes.object.isRequired,
+  };
+
+  getEventView(): EventView {
+    const {location} = this.props;
+
+    return EventView.fromLocation(location);
+  }
+
+  getEndpoints(): Array<[string, string, {query: EventQuery}]> {
+    const {organization, params, location} = this.props;
+    const {eventSlug} = params;
+    const eventView = this.getEventView();
+
+    const query = eventView.getEventsAPIPayload(location);
+
+    const url = `/organizations/${organization.slug}/events/${eventSlug}/`;
+
+    // Get a specific event. This could be coming from
+    // a paginated group or standalone event.
+    return [['event', url, {query}]];
+  }
+
+  get projectId() {
+    return this.props.eventSlug.split(':')[0];
+  }
+
+  renderBody() {
+    const {event} = this.state;
+
+    if (!event) {
+      return this.renderWrapper(<NotFound />);
+    }
+
+    return this.renderWrapper(this.renderContent(event));
+  }
+
+  renderContent(event: Event) {
+    const {organization, location} = this.props;
+    const eventView = this.getEventView();
+
+    if (!event) {
+      return this.renderWrapper(<NotFound />);
+    }
```

Comment:
Do you need this if renderBody() also checks the event?

Reply:
Good catch. Remove in https://github.com/getsentry/sentry/pull/15562/commits/f634b2410197456b2d0e4a6c2e2e2e462f8f3fb5